### PR TITLE
fix(#1520) file type changes are not detected.

### PR DESCRIPTION
**Pull request by sebcode**
Potential fix for #1520.

### CODE REVIEW

File: lua/nvim-tree/explorer/reload.lua [hunk 1]
```diff
@@ -45,6 +54,20 @@ function M.reload(node, status)
...
       child_names[abs] = true
+
+      -- Recreate node if type changes.
+      if nodes_by_path[abs] then
+        local n = nodes_by_path[abs]
+
+        if n.type ~= t then
+          local idx = table_find(node.nodes, n)
+          if idx then
+            table.remove(node.nodes, idx)
```
Comment (alex-courtis) [inline lua/nvim-tree/explorer/reload.lua:65]: We need to destroy the node here: `common.node_destroy(n)`

File: lua/nvim-tree/explorer/reload.lua [hunk 2]
```diff
@@ -18,6 +18,15 @@ local function update_status(nodes_by_path, node_ignored, status)
   end
 end
 
+local function table_find(table, value)
```
Comment (alex-courtis) [inline lua/nvim-tree/explorer/reload.lua:21]: This is useful. Please move to utils:

```lua
function M.array_index_of(array, item)
```

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Tested OK:
file -> link
file -> dir
link -> file
link -> dir
dir -> file
dir -> link

This will fit nicely into some ongoing work to change nodes to classes.

- [ ] utils function
- [ ] destory node

File: lua/nvim-tree/explorer/reload.lua (hunk 2 above)
Comment (sebcode) [reply lua/nvim-tree/explorer/reload.lua]: Actually `utils.array_remove` is exactly what I need

**Review by alex-courtis — APPROVED** [review]
Re-tested OK

**Pull request merged**